src/convenience/keyboard.ts

```diff
@@ -257,6 +375,120 @@ export class Keyboard {
         this.input_field_placeholder = value;
         return this;
     }
+    /**
+     * Creates a new keyboard that contains the transposed grid of buttons of
+     * this keyboard. This means that the resulting keyboard has the rows and
+     * columns flipped.
+     *
+     * Note that buttons can only span multiple columns, but never multiple
+     * rows. This means that if the given arrays have different lengths, some
+     * buttons might flow up in the layout. In these cases, transposing a
+     * keyboard a second time will not undo the first transposition.
+     *
+     * Here are some examples.
+     *
+     * ```
+     * original    transposed
+     * [  a  ]  ~> [  a  ]
+     *
+     *             [  a  ]
+     * [a b c]  ~> [  b  ]
+     *             [  c  ]
+     *
+     * [ a b ]     [a c e]
+     * [ c d ]  ~> [ b d ]
+     * [  e  ]
+     *
+     * [ a b ]     [a c d]
+     * [  c  ]  ~> [ b e ]
+     * [d e f]     [  f  ]
+     * ```
+     */
+    toTransposed() {
+        const original = this.keyboard;
+        const transposed = transpose(original);
+        return this.clone(transposed);
+    }
+    /**
+     * Creates a new keyboard with the same buttons but wrapped after a given
+     * number of columns (default: 4) as if the buttons were text elements.
+     * Optionally, you can specify if the wrapping should prioritize filling up
+     * the top (default) or the bottom row.
+     *
+     * This method is idempotent, so calling it a second time will effectively
+     * clone this keyboard without reordering the buttons.
+     *
+     * Here are some examples.
+     *
+     * ```
+     * original    wrapped
+     * [  a  ]  ~> [  a  ]    (4 columns)
+     *
+     *             [  a  ]
+     * [a b c]  ~> [  b  ]    (1 column)
+     *             [  c  ]
+     *
+     * [ a b ]     [a b c]
+     * [ c d ]  ~> [ d e ]    (3 columns)
+     * [  e  ]
+     *
+     * [ a b ]     [abcde]
+     * [  c  ]  ~> [  f  ]    (5 columns)
+     * [d e f]
+     *
+     * [a b c]     [  a  ]
+     * [d e f]  ~> [b c d]    (3 colums, bottom wrapping)
+     * [g h i]     [e f g]
+     * [  j  ]     [h i j]
+     * ```
+     *
+     * @param columns Maximum number of buttons per row
+     * @param wrap Optional wrapping behavior
+     */
+    toWrapped(columns = 4, wrap: "top" | "bottom" = "top") {
```

Comment:
Why the default value?

Reply:
when I first saw it I thought its overly complex and `bottom` is probably never needed :D so having `top` the default is better than specifying it every time

Reply:
Are you suggesting the value is bad or that we shouldn't have a default at all?

Reply:
i think that wrap: top/bottom is not intuitive by itself. you can't tell what it does before you look at the examples. it could very well be the opposite behavior

Reply:
I think `columns` shouldn't have a default